File: SwordShopBasicCrud/Models/Order.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SwordShopBasicCrud.Models
{
    public class Order : BaseModel
    {
        public DateTime Date { get; set; }
        public decimal Value { get; set; }
        public decimal Discount { get; set; }
        public decimal Total { get; set; }

        public long ClientId { get; set; }
        public virtual Client Client { get; set; }

        public virtual IList<Product> Products { get; set; }
    }
}

File: Front/src/app/core/models/order.ts
import { BaseModel } from './base-model';
import { Client } from './client';
import { Product } from './product';

export class Order extends BaseModel {
  date?: string;
  value?: number;
  discount?: number;
  total?: number;

  clientId?: number;
  client?: Client;

  products?: Product[];
}
